src/Ouroboros-Mini_Protocols-Chain_Sync.thy

```diff
@@ -0,0 +1,226 @@
+section \<open>Chain-Sync Mini-Protocol\<close>
+
+text \<open>
+  The theory in this section implements the chain synchronization mini-protocol.
+\<close>
+
+theory "Ouroboros-Mini_Protocols-Chain_Sync"
+  imports
+    "Ouroboros-Mini_Protocols"
+    "HOL-Library.BNF_Corec"
+begin
+
+subsection \<open>Cryptographic Primitives\<close>
+
+typedecl hash
+
+axiomatization
+where
+  hash_linear_order: "OFCLASS(hash, linorder_class)"
+
+instance hash :: linorder
+  by (fact hash_linear_order)
```

Comment:
We should avoid `typedecl` with `axiomatization` as much as possible. As you know, `axiomatization` can introduce inconsistencies. Furthermore, it prevents us from interpreting an abstract specification in different concrete ways. I think the way to go here is really to use a locale.

It is true that for Þ-calculus channels I used type declarations and axiomatizations, but I did so only because the type for typed channels is parameterized and we cannot have higher-order type parameters.

Regarding the linear-order constraint: Do we really need this? At the moment, it seems to me that all we need is having a type parameter for hashes. This is something we could just add to the `chain_sync` locale, I guess.